src/npctalk_rules.cpp

```diff
@@ -0,0 +1,467 @@
+                                             role_model.rules.cbm_reserve == guy->rules.cbm_reserve &&
+                                             role_model.rules.pickup_whitelist->empty() == guy->rules.pickup_whitelist->empty() );
+                /*The great swap*/
+                // To make this slightly easier...
+                npc *guy_with_same = exporting_rules ? guy : &role_model;
+                npc *guy_with_new = exporting_rules ? &role_model : guy;
+                if( do_flags_overrides || do_all ) {
+                    guy_with_new->rules.flags = guy_with_same->rules.flags;
+                    guy_with_new->rules.overrides = guy_with_same->rules.overrides;
+                    guy_with_new->rules.override_enable = guy_with_same->rules.override_enable;
+                }
+                if( do_aim || do_all ) {
+                    guy_with_new->rules.aim = guy_with_same->rules.aim;
+                }
+                if( do_engagement || do_all ) {
+                    guy_with_new->rules.engagement = guy_with_same->rules.engagement;
+                }
+                if( do_cbm_recharge || do_all ) {
+                    guy_with_new->rules.cbm_recharge = guy_with_same->rules.cbm_recharge;
+                }
+                if( do_cbm_reserve || do_all ) {
+                    guy_with_new->rules.cbm_reserve = guy_with_same->rules.cbm_reserve;
+                }
+                if( do_pickup || do_all ) {
+                    guy_with_new->rules.pickup_whitelist = guy_with_same->rules.pickup_whitelist;
+                }
+            }
+        }
+        ImGui::EndTable();
+    }
+}
+
+void follower_rules_ui_impl::draw_controls()
+{
+    if( !guy ) {
+        debugmsg( "Something has gone very wrong, can't find NPC to set rules for.  Aborting." );
+        last_action = "QUIT";
+        return;
+    }
+
+    // ImGui only captures arrow keys by default, we want to also capture the numpad and hjkl
+    if( last_action == "QUIT" ) {
+        return;
+    } else if( last_action == "UP" ) {
+        ImGui::NavMoveRequestSubmit( ImGuiDir_Up, ImGuiDir_Up, ImGuiNavMoveFlags_None,
+                                     ImGuiScrollFlags_None );
+    } else if( last_action == "DOWN" ) {
+        ImGui::NavMoveRequestSubmit( ImGuiDir_Down, ImGuiDir_Down, ImGuiNavMoveFlags_None,
+                                     ImGuiScrollFlags_None );
+    } else if( last_action == "LEFT" ) {
+        ImGui::NavMoveRequestSubmit( ImGuiDir_Left, ImGuiDir_Left, ImGuiNavMoveFlags_None,
+                                     ImGuiScrollFlags_None );
+    } else if( last_action == "RIGHT" ) {
+        ImGui::NavMoveRequestSubmit( ImGuiDir_Right, ImGuiDir_Right, ImGuiNavMoveFlags_None,
+                                     ImGuiScrollFlags_None );
+    } else if( last_action == "CONFIRM" ) {
+        ImGui::ActivateItem( ImGui::GetFocusID() );
+    }
```

Comment:
Oops, I just noticed this bug. The Button handles the enter key itself, so this code double–activates the button. When you press enter the setting will flicker as it is toggled on and then immediately back off. You can just remove this clause.

Reply:
It does not handle numpad enter.

Reply:
Awkward.

Reply:
Marking as resolved, then.